synapse/handlers/user_directory.py

```diff
@@ -132,12 +132,7 @@ async def handle_local_profile_change(
         # FIXME(#3714): We should probably do this in the same worker as all
         # the other changes.
 
-        # Support users are for diagnostics and should not appear in the user directory.
-        is_support = await self.store.is_support_user(user_id)
-        # When change profile information of deactivated user it should not appear in the user directory.
-        is_deactivated = await self.store.get_user_deactivated_status(user_id)
-
-        if not (is_support or is_deactivated):
+        if not await self.store.is_excluded_from_user_dir(user_id):
```

Comment:
I find the double-negative here a bit mind-bending. Perhaps we should instead have an `should_include_in_user_dir` ?

Reply:
Glad it's not just me. I was erring towards preserving the original logic below as much as possible, but think it'll be better this way round.